project/ParadoxSupport.scala

```diff
@@ -29,23 +30,31 @@ object ParadoxSupport {
 
   class UnidocDirective(allClasses: IndexedSeq[String]) extends InlineDirective("unidoc") {
     def render(node: DirectiveNode, visitor: Visitor, printer: Printer): Unit = {
-      if (node.label.split('[')(0).contains('.')) {
-        val fqcn = node.label
+      val (directive, label, source) = node.source match {
+        case direct: Source.Direct => (s"@unidoc[${node.label}](${direct.value})", node.label, direct.value)
+        case ref: Source.Ref       => (s"@unidoc[${node.label}](${ref.value})", node.label, ref.value)
+        case Source.Empty          => (s"@unidoc[${node.label}]", node.label.split('.').last, node.label)
+      }
+
+      if (source.split('[')(0).contains('.')) {
+        val fqcn = source
         if (allClasses.contains(fqcn)) {
-          val label = fqcn.split('.').last
           syntheticNode("java", javaLabel(label), fqcn, node).accept(visitor)
-          syntheticNode("scala", label, fqcn, node).accept(visitor)
+          syntheticNode("scala", scalaLabel(label), fqcn, node).accept(visitor)
         } else {
-          throw new java.lang.IllegalStateException(s"fqcn not found by @unidoc[$fqcn]")
+          throw new java.lang.IllegalStateException(s"fqcn not found by $directive")
```

Comment:
`$directive` is what users put in .md file, (i.e.) `@unidoc[****](****)` or `@unidoc[*****]`